localstack/services/awslambda/provider.py

```diff
@@ -1346,6 +1347,13 @@ def create_event_source_mapping(
         if "EventSourceArn" not in request:
             raise InvalidParameterValueException("Unrecognized event source.", Type="User")
 
+        service = extract_service_from_arn(request.get("EventSourceArn"))
+        if "StartingPosition" not in request and service in ["dynamodb", "kinesis", "kafka"]:
```

Comment:
```suggestion
        if service in ["dynamodb", "kinesis", "kafka"] and "StartingPosition" not in request:
```

nit: IMO, it's a bit more natural to read this way :thinking: 



Also, I think this should also be evaluated in update_event_source_mapping.

We'll need to extend these validations, especially for event source mappings at some point even further, since there's a lot of differences in which sets of values are valid depending on the source service.